geometry/named_quantities.hpp

```diff
@@ -10,6 +11,27 @@
 namespace principia {
 namespace geometry {
 
+// A trait to treat types that have a norm uniformly (using Abs for quantities
+// or double, and Norm for multivectors).
+template<typename T,
+         typename =
+             base::void_if_exists<decltype(quantities::Abs(std::declval<T>()))>>
+struct Normed : base::not_constructible {
+  using NormType = T;
+  static NormType Norm(T const& vector) {
+    return quantities::Abs(vector);
```

Comment:
Please move the method implementations to the body file.

Reply:
I don't know to write the definitions out of line with this fancy specialization.

Reply:
See https://godbolt.org/g/GXy9jk.

Reply:
Done.